include/rdma/fi_domain.h

```diff
@@ -237,6 +237,22 @@ static inline uint64_t fi_mr_key(struct fid_mr *mr)
 	return mr->key;
 }
 
+static inline int
+fi_mr_raw_key(struct fid_mr *mr, uint8_t *key, size_t *size)
+{
+	struct fi_mr_raw_key lakey = {.size = size, .key = key};
+	return mr->fid.ops->control(&mr->fid, FI_RAWKEY, &lakey);
+}
+
+static inline uint64_t
+fi_mr_map_key(struct fid_domain *domain, uint8_t *key, size_t size)
+{
+	struct fi_mr_map_raw_key lakey =
+		{.size = size, .key = key, .mapkey = 0};
+	domain->fid.ops->control(&domain->fid, FI_RAWKEY, &lakey);
+	return lakey.mapkey;
+}
```

Comment:
I think we would need to change the call notation to that.  I like the version that you gave, but I'm worried that we would lose error information.